### PR TITLE
Second draft revisions

### LINKED ISSUE #3: Remove redundant "Client" in scheme?
The "Client" suffixes for the scheme entries look like a implementation detail and is rather redundant. Maybe remove it?

### LINKED ISSUE #4: Key of the security property / De-duplication
The spec doesn't say anything about how the key for the security property should look like or what it's useful for. If it's just an arbitrary key, would it be better to specify it as an array?

Also, a minor thing. The security property says "Object that desribes the authenticated scheme and **href**", but the href is not specified in the object. Just description and scheme.

### LINKED ISSUE #8: Re-use OpenAPI?
A similar thing is defined in OpenAPI:
https://swagger.io/docs/specification/authentication/

Could we learn from it? Base this on top of it and extend it? ...

### LINKED ISSUE #10: Allow schemes other than the ones we know about
There will be schemes that aren't supported by **stac-asset**, and more generally, we can't assume to know all the different ways folks might secure their assets. It seems like some sort of fallback would be helpful, e.g. `other`. Alternatively, we could allow anything in that field, so folks could define their own keys and behavior (e.g. `my_special_authentication_scheme`).

### PR DESCRIPTION
**Pull request by jamesfisher-geo**
Changes to align with the OpenAPI authorization spec and to remove deduplication by defining schema definitions in the Item/Collection properties.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,93 +24,129 @@ API to generate signed URLs. See the [Signed URL](#url-signing) section for an e
...
+| `earthdata`               | Uses a token-based authentication to download data, from *some* Earthdata providers, e.g. DAACs                          |
+| `signedUrl`               | Signs URLs with a user-defined authentication API                                                                        |
+| `oauth2`                  | [Open Authentication 2.0](https://swagger.io/docs/specification/authentication/oauth2/) configuration                                                                                                                                          |
+| `apiKey`                  | Description of [API key](https://swagger.io/docs/specification/authentication/api-keys/) authentication included in request headers, query parameters, or cookies.                                                                                                                                               |
+| `openIdConnect`           | Description of [OpenID Connect Discovery](https://swagger.io/docs/specification/authentication/openid-connect-discovery/) authentication
```
Comment (gadomski) [inline README.md]: Many STAC keys are `snake_case` (e.g. `start_datetime`) -- is the choice to use `camelCase` here based on OpenAPI? If it's not, I'd suggest switching to `snake_case`, e.g. `planetary_computer`.

File: examples/item.json [hunk 2]
```diff
@@ -39,7 +39,28 @@
...
+          "type": "oauth2",
+          "description": "requires a login and user token",
+          "flows": {
+                 "authorizationUrl": "https://example.com/oauth/authorize",
+                 "tokenUrl": "https://example.com/oauth/token",
+              "scopes": {
+                "read:example": "Read the example data",
+                "write:example": "Write the example data",
+                "admin:example": "Read/write/delete the example data"
+              }
```
Comment (gadomski) [inline examples/item.json]: Formatting seems off here -- we should have some sort of auto-formatter as a part of CI to enforce common style.

File: README.md [hunk 3]
```diff
@@ -26,93 +24,129 @@ API to generate signed URLs. See the [Signed URL](#url-signing) section for an e
...
 ### URL Signing
 
-The `SignedUrlClient` scheme indicates that authentication will be handled by an API, which generates and returns a signed URL. For example, 
-a signed URL for an asset in AWS S3 can be generated with the following Lambda function code.
+The `signedUrl` scheme indicates that authentication will be handled by an API which generates and returns a signed URL. A signed URL authentication scheme can be defined with 
+```json
+{
+  "security:schemes": {
+    "signedUrl": {
+      "type": "signedUrl",
```
Comment (gadomski) [inline README.md]: This feels awkward, having the key be the same as the type. If we want to support multiple schemes with the same scheme (e.g. two signed URL options), then `security:schemes` should probably become a list?

Comment (jamesfisher-geo) [reply README.md]: Agreed. The key can be any string. Perhaps `signedUrlAuth` or `signed_url_auth` are better.

Comment (gadomski) [reply README.md]: What's the point of the key again? Seems like if someone's trying to fetch a specific security scheme object, they're most likely to fetch by the scheme's name. Is there a reason to _not_ do a list?

File: README.md (hunk 1 above)
Comment (jamesfisher-geo) [reply README.md]: The `camelCase` here is based on OpenAPI. However, I think we can switch to `snake_case` to align with the STAC.

File: README.md (hunk 3 above)
Comment (jamesfisher-geo) [reply README.md]: The keys are used in the `security:refs` asset property to reference an auth scheme.

The thought behind using a dictionary is that allows multiple schemes of the same type. An ordered list could do this as well. But a dictionary seems more readable.


```json
"security:schemes": {
      "api_auth_1": {
          "type": "apiKey",
          "description": "Requires an API key in the request header",
          "in": "header",
          "name": "x-api-key"
          "flows": {}
          },
      "api_auth_2": {
          "type": "apiKey",
          "description": "Requires an API key in the request query parameter",
          "in": "query",
          "name": "key"
          "flows": {}
          }
    }
```

```json
"assets": {
    "api_examples_1": {
      "href": "https://example1.com/examples/1",
      "title": "Secure Asset Example",
      "type": "application/vnd.example",
      "roles": [
        "data"
      ],
      "security:refs": [
        "api_auth_1"
      ]
    },
   "api_examples_2": {
      "href": "https://example2.com/examples/2",
      "title": "Secure Asset Example",
      "type": "application/vnd.example",
      "roles": [
        "data"
      ],
      "security:refs": [
        "api_auth_2"
      ]
    }
  }
```

Comment (gadomski) [reply README.md]: Ok yup the `security:refs` is the reason I forgot. Thanks for reminding me.

**Pull request merged**